### PR TITLE
fix account switcher and address component

### PR DESCRIPTION
**Pull request by turbocrime**
this was causing rendering errors and strange behavior in parent components. it should now render more reliably

for some reason it's showing henry as 'collaborating', but that's not quite true. this came out of some work to use this component and make its style configurable for light theme, in my ibc style branch

### CODE REVIEW

File: packages/ui/components/ui/address-component.tsx [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+  ephemeral,
+  ephemeralClassName = 'text-rust',
+}: AddressComponentProps) => {
   const bech32Addr = bech32Address(address);
 
   return (
-    <span
-      className={'font-mono' + (ephemeral ? ' text-[#8D5728]' : ' text-muted-foreground truncate')}
-    >
+    <div className={cn('font-mono', 'truncate w-0 min-w-full', ephemeral && ephemeralClassName)}>
```
Comment (turbocrime) [inline packages/ui/components/ui/address-component.tsx:23]: now inherits text color from parent, and 'alternate' color is configurable

File: packages/ui/components/ui/select-account.tsx [hunk 2]
```diff
@@ -33,52 +33,43 @@ export const SelectAccount = ({ getAddrByIndex }: SelectAccountProps) => {
...
-              </div>
-
-              <p className='truncate text-sm'>
-                <AddressComponent address={address} ephemeral={ephemeral} />
-              </p>
-            </div>
-            <CopyToClipboardIconButton text={bech32Address(address)} />
+        <div className='mt-4 flex items-center justify-between gap-1 rounded-lg border bg-background px-3 py-4'>
+          <AddressIcon address={address} size={24} />
+          <div className='w-full text-muted-foreground'>
```
Comment (turbocrime) [inline packages/ui/components/ui/select-account.tsx:42]: the old default text color `text-muted-foreground` is applied at existing use locations

File: apps/minifront/src/components/dashboard/assets-table.tsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ export default function AssetsTable() {
                 </h2>
               </div>
 
-              <div className='max-w-72 truncate'>
+              <div className='max-w-72 text-muted-foreground'>
                 <AddressComponent address={a.address} />
               </div>
             </div>
```
Comment (grod220) [inline apps/minifront/src/components/dashboard/assets-table.tsx:47]: We appear to have lost the address part, before:

Screenshot 2024-03-28 at 10 28 55 AM

After:
Screenshot 2024-03-28 at 10 28 42 AM

File: packages/ui/components/ui/select-account.tsx [hunk 4]
```diff
@@ -33,52 +33,43 @@ export const SelectAccount = ({ getAddrByIndex }: SelectAccountProps) => {
...
-                <AddressIcon address={address} size={24} />
-              </div>
-
-              <p className='truncate text-sm'>
-                <AddressComponent address={address} ephemeral={ephemeral} />
-              </p>
-            </div>
-            <CopyToClipboardIconButton text={bech32Address(address)} />
+        <div className='mt-4 flex items-center justify-between gap-1 rounded-lg border bg-background px-3 py-4'>
+          <AddressIcon address={address} size={24} />
```
Comment (grod220) [inline packages/ui/components/ui/select-account.tsx:41]: The identicon looks to have lost some roundness here: 
Screenshot 2024-03-28 at 10 31 08 AM

Comment (turbocrime) [reply packages/ui/components/ui/select-account.tsx:41]: this is a failure in identicon, have a fix that isn't PR'd yet

**Pull request closed without merging**